File: JavaScript_Courses/JavaScript_Basics/Preparation/02_Conditional_Statements_Exercise/06_World_Swimming_Record.js
function worldSwimmingRecord(input) {
    let timeLostPer15Meters = 12.5;

    let recordTime = Number(input[0]);
    let meters = Number(input[1]);
    let secondsPerMeter = Number(input[2]);

    let numberOfLosts = Math.floor(meters / 15);
    let lostTime = numberOfLosts * timeLostPer15Meters;

    let totalTime = secondsPerMeter * meters + lostTime;

    if (totalTime < recordTime) {
        console.log(`Yes, he succeeded! The new world record is ${totalTime.toFixed(2)} seconds.`);
    } else {
        console.log(`No, he failed! He was ${(totalTime - recordTime).toFixed(2)} seconds slower.`);
    }
}

// worldSwimmingRecord(["10464", "1500", "20"]);
// worldSwimmingRecord(["55555.67", "3017", "5.03"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/02_Conditional_Statements_Exercise/06_World_Swimming_Record.py
# import math

record_seconds = float(input())
distance_in_meters = float(input())
seconds_per_meter = float(input())

# slowdown_seconds = math.floor(distance_in_meters / 15) * 12.5
slowdown_seconds = distance_in_meters // 15 * 12.5
swimming_seconds = distance_in_meters * seconds_per_meter

total_seconds = swimming_seconds + slowdown_seconds

if total_seconds < record_seconds:
    print(f'Yes, he succeeded! The new world record is {total_seconds :.2f} seconds.')
else:
    diff = total_seconds - record_seconds
    print(f'No, he failed! He was {diff :.2f} seconds slower.')
